internal/edgedbtypes/uuid.go

```diff
@@ -22,6 +22,24 @@ import (
 	"strings"
 )
 
+// ParseUUID parses s into a UUID or returns an error.
+func ParseUUID(s string) (UUID, error) {
+	s = strings.Replace(s, "-", "", 4)
+
+	var id UUID
+	for i := 0; i < 16; i++ {
+		val, err := strconv.ParseUint(s[:2], 16, 8)
```

Comment:
Perhaps a faster way would be to manually unroll this and parse it as two int64. You can easily break them into individual bytes after.

Reply:
If we are looking for optimal performance maybe we should do something like [this](https://github.com/google/uuid/blob/bfb86fa49a73e4194d93bea18d7acfe3694438ce/uuid.go#L84)?

Reply:
Oh wow, they went to great length to avoid copying that string.  Up to you, actually.  I'm fine even if you merge this PR as is.